test/unit_test/quartz/physics/rigid_body/test_rigidbody_collider_pointers.cpp

```diff
@@ -0,0 +1,243 @@
+#include "quartz/managers/physics_manager/PhysicsManager.hpp"
+#include "quartz/physics/collider/Collider.hpp"
+#include "quartz/physics/field/Field.hpp"
+#include "quartz/physics/rigid_body/RigidBody.hpp"
+
+#include "reactphysics3d/body/RigidBody.h"
+#include "reactphysics3d/components/RigidBodyComponents.h"
+#include "reactphysics3d/engine/PhysicsWorld.h"
+#include "unit_test/Util.hpp"
+#include "util/logger/Logger.hpp"
+
+int test_collider_from_constructor_mbody_pointer() {
+    LOG_FUNCTION_SCOPE_INFO(UNIT_TEST, "");
+    int result = 0;
+
+    // create the physics field so we can give it to the doodad's constructor
+    LOG_TRACE(UNIT_TEST, "Creating physics manager");
+    quartz::managers::PhysicsManager& physicsManager = quartz::unit_test::UnitTestClient::getPhysicsManagerInstance();
+    reactphysics3d::PhysicsWorld::WorldSettings physicsWorldSettings;
+    reactphysics3d::PhysicsWorld* p_physicsWorld = physicsManager.createPhysicsWorldPtr(physicsWorldSettings);
+    UT_REQUIRE(p_physicsWorld);
+
+    // create the inputTransform that we are using for the doodad
+    LOG_TRACE(UNIT_TEST, "Creating transform");
+    const math::Vec3 position(42.0f, 69.0f, -100.0f);
+    const math::Quaternion rotation = math::Quaternion::fromAxisAngleRotation({1.0f, 0.0f, 0.0f}, 45.0f);
+    const math::Vec3 scale(1.0f, 3.0f, 5.0f);
+    const math::Transform inputTransform(position, rotation, scale);
+
+    // create reactphysics3d::RigidBody*
+    LOG_TRACE(UNIT_TEST, "Creating rigid body pointer");
+    reactphysics3d::RigidBody* p_rigidBody = quartz::physics::RigidBody::createRigidBodyPtr(
+        p_physicsWorld,
+        inputTransform,
+        reactphysics3d::BodyType::DYNAMIC,
+        true,
+        {0, 0, 0}
+    );
+    UT_REQUIRE(p_rigidBody);
+
+    // box collider parameters
+    LOG_TRACE(UNIT_TEST, "Creating box collider parameters");
+    const quartz::physics::BoxShape::Parameters boxShapeParameters({1.0f, 1.0f, 1.0f});
+
+    // creating box shape pointer
+    LOG_TRACE(UNIT_TEST, "Creating box shape pointer");
+    reactphysics3d::BoxShape* p_boxShape = physicsManager.createBoxShapePtr(boxShapeParameters.halfExtents);
+    UT_REQUIRE(p_boxShape);
+
+    // creating collider pointer
+    LOG_TRACE(UNIT_TEST, "Creating collider pointer");
+    reactphysics3d::Collider* p_collider = quartz::physics::Collider::createColliderPtr(p_rigidBody, p_boxShape);
+    UT_REQUIRE(p_collider);
```

Comment:
is there a nice way to pull all of this out into a helper function so we don't need to be duplicating the code all the time? Perhaps we can just be less verbose about everything and reduce the line count that way (less comments, less intermediate variables, less print statements, etc)